src/client/params.rs

```diff
@@ -199,3 +201,39 @@ pub enum PhyFlag {
     LECodedTx = 1 << 13,
     LECodedRx = 1 << 14,
 }
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, FromPrimitive)]
+#[repr(u16)]
+pub enum SystemConfigParameterType {
+    BREDRPageScanType = 0x0000,
+    BREDRPageScanInterval,
+    BREDRPageScanWindow,
+    BREDRInquiryScanType,
+    BREDRInquiryScanInterval,
+    BREDRInquiryScanWindow,
+    BREDRLinkSupervisionTimeout,
+    BREDRPageTimeout,
+    BREDRMinSniffInterval,
+    BREDRMaxSniffInterval,
+    LEAdvertisementMinInterval,
+    LEAdvertisementMaxInterval,
+    LEMultiAdvertisementRotationInterval,
+    LEScanningIntervalForAutoConnect,
+    LEScanningWindowForAutoConnect,
+    LEScanningIntervalForWakeScenarios,
+    LEScanningWindowForWakeScenarios,
+    LEScanningIntervalForDiscovery,
+    LEScanningWindowForDiscovery,
+    LEScanningIntervalForAdvMonitoring,
+    LEScanningWindowForAdvMonitoring,
+    LEScanningIntervalForConnect,
+    LEScanningWindowForConnect,
+    LEMinConnectionInterval,
+    LEMaxConnectionInterval,
+    LEConnectionLatency,
+    LEConnectionSupervisionTimeout,
+    LEAutoconnectTimeout,
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, FromPrimitive)]
+pub enum RuntimeConfigParameterType {}
```

Comment:
It seems like this type is going to be extended in later versions of the Linux kernel, so maybe add a `#[non_exhaustive]`?

Reply:
Also, why does this type not have `#[repr(u16)]`? Is this not allowed on empty enums?

Reply:
Yes, unfortunately it's not allowed. Perhaps it makes more sense to leave the whole thing out until there are known variants introduced in the API?